service/routes.py

```diff
@@ -115,6 +115,26 @@ def get_shopcarts(shopcart_id):
     app.logger.info("Returning shopcart: %s", shopcart.id)
     return make_response(jsonify(shopcart.serialize()), status.HTTP_200_OK)
 
+
+######################################################################
+# DELETE A SHOPCART
+######################################################################
+@app.route("/shopcarts/<int:shopcart_id>", methods=["DELETE"])
+def delete_shopcart(shopcart_id):
+    """Deletes a shopcart
+        Args:
+            user_id (str): the user_id of the shopcart to delete
+        Returns:
+            str: always returns an empty string
+    """
+    app.logger.info("Start deleting shopcart %s...", shopcart_id)
+    shopcart = Shopcart.get_by_id(shopcart_id)
+    if shopcart:
```

Comment:
we need to return 404 Not Found when shopcart_id does not exist

Reply:
also, you would need a corresponding test case like [what @momo100110 Kangle did](https://github.com/CSCI-GA-2820-SU23-001/shopcarts/blob/master/tests/test_routes.py#L556) for the Delete Shopcart Item API to remove line  133-134 from lines missing test cases.

Reply:
🤔 According to lecture slide 03-Building-RESTful-Services page43, does delete always return 204_NO_CONTENT?

Reply:
I double checked the slide, and you're right. Sorry to confuse you.

Reply:
This reminds me that the Delete Shopcart API doc needs to be revised accordingly, please help to update the doc, thanks!

Reply:
So I need to modify here in Delete A Shopcart Item API as well, right? @nijya @wl2484

Reply:
> 

Yes